clients/java/signalr/build.gradle

```diff
@@ -16,3 +17,32 @@ dependencies {
     implementation "org.java-websocket:Java-WebSocket:1.3.8"
     implementation 'com.google.code.gson:gson:2.8.5'
 }
+
+task sourceJar(type: Jar) {
+    classifier "sources"
+    from sourceSets.main.allJava
+}
+
+task javadocJar(type: Jar, dependsOn: javadoc) {
+    classifier "javadoc"
+    from javadoc.destinationDir
+}
+
+task generatePOM {
+    pom {
+        project {
+            groupId 'com.microsoft.aspnetcore'
+            artifactId 'signalr'
+            version '0.1.0'
+
+            inceptionYear '2018'
+            licenses {
+                license {
+                    name 'The Apache Software License, Version 2.0'
+                    url 'http://www.apache.org/licenses/LICENSE-2.0.txt'
+                    distribution 'repo'
+                }
+            }
+        }
+    }.writeTo("pom.xml")
```

Comment:
Do we want to give this a different name/put this in a different place?

Reply:
Call it `signalr-client-0.1.0-preview1.pom` (like the JAR files) and have the gradle build drop it into the `build/libs` directory along side them (so we have a directory that can just be copied over to `artifacts`. Then, add an MSBuild [Copy](https://msdn.microsoft.com/en-us/library/3e54c37h.aspx) Task after the `Exec` tasks to copy them to `$(BuildDir)`.

That should cause the build to start failing due to "undeclared artifacts", which is good, it means the artifacts are in the right place. Then you just need to define `ArtifactInfo` MSBuild items for them. @natemcmaster can help out a bit there too.

Take a look at the `GetNpmArtifactInfo` target. It's wired in to the `GetArtifactInfo` phase that KoreBuild defines. We'll want a `GetJavaArtifactInfo` target that works the same I think. I'm not sure if KoreBuild will need any updates to handle that, since we'll want to use a new ArtifactType.

Reply:
> Call it signalr-client-0.1.0-preview1.pom (like the JAR files

I think this might actually need to be named `pom.xml`

Reply:
It's required to be `pom.xml` when you're using Maven to build, but we're using Gradle. When publishing to Maven Central it should be `signalr-client-0.1.0-preview1.pom`. The [Maven Central Requirements doc](https://central.sonatype.org/pages/requirements.html) seems to indicate that in the examples.

Reply:
Ah okay cool. I was looking at some different sources. That link is very clear 👍

Reply:
> Yes, please define an <ArtifactInfo> item with ArtifactType="ICanHasJavaPackageStuff"

I suggest `MavenPomFile` for the `.pom` and `JavaJar` for the `.jar` files.